h/api/test/uri_test.py

```diff
@@ -154,6 +153,13 @@
     ("http://example.com?gclid_foo=abcde", "http://example.com?gclid_foo=abcde"),
     ("http://example.com?bar_gclid=abcde", "http://example.com?bar_gclid=abcde"),
     ("http://example.com?WT=abcde", "http://example.com?WT=abcde"),
-])
+]
+
+
+@pytest.mark.parametrize("url_in,url_out", TEST_URLS)
 def test_normalize(url_in, url_out):
     assert uri.normalize(url_in) == url_out
+
+@pytest.mark.parametrize("url,_", TEST_URLS)
```

Comment:
Rather than duplicating the tests, I'd suggest you add `from __future__ import unicode_literals` to the top of the test file, and then add a specific test to ensure that URLs get mapped to text:

```
@pytest.mark.parametrize("url_in", [
    "http://example.com/text",
    b"http://example.com/bytes",
])
def test_normalize_returns_text(url_in):
    assert isinstance(uri.normalize(url_in), text_type)
```

And note that you should use `text_type`, which is `str` on Python 3 and `unicode` on Python 2. I'd suggest creating a `h/api/_compat.py` and copying the relevant parts of `h/_compat.py` into it.